src/assets/scripts/airport/Arrival/ArrivalBase.js

```diff
@@ -61,150 +187,208 @@ export default class ArrivalBase {
             this.heading = degreesToRadians(options.heading);
         }
 
+        // TODO: is altitude ever not a number?
         if (typeof this.altitude === 'number') {
             this.altitude = [this.altitude, this.altitude];
         }
 
         if (options.route) {
-            this.route = options.route;
+            this.activeRouteModel = new RouteModel(options.route);
         } else if (options.fixes) {
-            for (let i = 0; i < options.fixes.length; i++) {
-                this.fixes.push({ fix: options.fixes[i] });
-            }
+            // TODO: this may not be needed at all. we could just use `_get()` instead.
```

Comment:
Agreed. In fact, the whole fix-array deal could theoretically just be removed. If users wanted their airports to just have arrival streams follow a string of fixes, there's no reason to not just make it a `route`, using dot-notation, and then have that formatted into a `RouteModel` just the same as if it had airways or STARs or any other procedures built into it.